src/framework/components/anim/controller.js

```diff
@@ -0,0 +1,662 @@
+                        for (j = 0; j < state.animations.length; j++) {
+                            animation = state.animations[j];
+                            this._animEvaluator.findClip(animation.name + '.previous.' + i).blendWeight = (1.0 - interpolatedTime) * animation.weight / state.totalWeight * stateWeight;
+                        }
+                    }
+                    // while transitioning, set active state animations to be weighted by (interpolationTime).
+                    state = this.activeState;
+                    for (i = 0; i < state.animations.length; i++) {
+                        animation = state.animations[i];
+                        this._animEvaluator.findClip(animation.name).blendWeight = interpolatedTime * animation.weight / state.totalWeight;
+                    }
+                } else {
+                    this._isTransitioning = false;
+                    // when a transition ends, remove all previous state clips from the evaluator
+                    var activeClips = this.activeState.animations.length;
+                    var totalClips = this._animEvaluator.clips.length;
+                    for (i = 0; i < totalClips - activeClips; i++) {
+                        this._animEvaluator.removeClip(0);
+                    }
+                    this._transitionPreviousStates = [];
+                    // when a transition ends, set the active state clip weights so they sum to 1
+                    state = this.activeState;
+                    for (i = 0; i < state.animations.length; i++) {
+                        animation = state.animations[i];
+                        this._animEvaluator.findClip(animation.name).blendWeight = animation.weight / state.totalWeight;
+                    }
+                }
+                this._currTransitionTime += dt;
+            }
+            this._animEvaluator.update(dt);
+        },
+
+        findParameter: function (name) {
+            return this._parameters[name];
+        }
+    });
+
+    return {
+        ANIM_INTERRUPTION_NONE: ANIM_INTERRUPTION_NONE,
+        ANIM_INTERRUPTION_PREV: ANIM_INTERRUPTION_PREV,
+        ANIM_INTERRUPTION_NEXT: ANIM_INTERRUPTION_NEXT,
+        ANIM_INTERRUPTION_PREV_NEXT: ANIM_INTERRUPTION_PREV_NEXT,
+        ANIM_INTERRUPTION_NEXT_PREV: ANIM_INTERRUPTION_NEXT_PREV,
+
+        ANIM_GREATER_THAN: ANIM_GREATER_THAN,
+        ANIM_LESS_THAN: ANIM_LESS_THAN,
+        ANIM_GREATER_THAN_EQUAL_TO: ANIM_GREATER_THAN_EQUAL_TO,
+        ANIM_LESS_THAN_EQUAL_TO: ANIM_LESS_THAN_EQUAL_TO,
+        ANIM_EQUAL_TO: ANIM_EQUAL_TO,
+        ANIM_NOT_EQUAL_TO: ANIM_NOT_EQUAL_TO,
+
+        ANIM_PARAMETER_INTEGER: ANIM_PARAMETER_INTEGER,
+        ANIM_PARAMETER_FLOAT: ANIM_PARAMETER_FLOAT,
+        ANIM_PARAMETER_BOOLEAN: ANIM_PARAMETER_BOOLEAN,
+        ANIM_PARAMETER_TRIGGER: ANIM_PARAMETER_TRIGGER,
+
+        ANIM_STATE_START: ANIM_STATE_START,
+        ANIM_STATE_END: ANIM_STATE_END,
```

Comment:
They could. I'm going to remove the `Object.assign(pc, {});` pattern from the engine in the next week or so. For now, let's go with this for consistency. But this will change.

Reply:
Or yeah, you could make it like graphics.js or framework\components\rigidbody\constants.js. I'm easy. Up to you @ellt92

Reply:
I've adopted the rigidbody style for this.